<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Введение" data-toc-modified-id="Введение-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Введение</a></span></li><li><span><a href="#Проверка-гипотез" data-toc-modified-id="Проверка-гипотез-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Проверка гипотез</a></span><ul class="toc-item"><li><span><a href="#Средние-пользовательские-рейтинги-платформ-XOne-и-PC-одинаковые" data-toc-modified-id="Средние-пользовательские-рейтинги-платформ-XOne-и-PC-одинаковые-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Средние пользовательские рейтинги платформ XOne и PC одинаковые</a></span></li><li><span><a href="#Cредние-пользовательские-рейтинги-жанров-Action-и-Sports-разные" data-toc-modified-id="Cредние-пользовательские-рейтинги-жанров-Action-и-Sports-разные-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Cредние пользовательские рейтинги жанров Action и Sports разные</a></span></li></ul></li><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Вывод</a></span></li></ul></div>

## Введение

Здесь проведена проверка двух гипотез по тем же данным, что находятся в основной тетрадке. Проверяется равность средних значений пользовательских оценок по двум компьютерным платформам и двум жанрам за 2016 год.

Ниже собраны данные, необходимые для тестов. Более детальный разбор данных представлен в основной части проекта.

In [29]:
import pandas as pd
from scipy import stats as st

In [30]:
try:
    data = pd.read_csv('games.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/games.csv') 

In [31]:
data.columns = data.columns.str.lower()

In [32]:
data = data.dropna(subset='year_of_release')
data['year_of_release'] = data['year_of_release'].astype(int)

In [33]:
data['user_score'] = data['user_score'].replace('tbd','nan')
data['user_score'] = data['user_score'].astype(float)

In [34]:
last_year_data = data.query('year_of_release==2016')

## Проверка гипотез

Для проверки гипотез мы используем t-тесты. 

Т-тест — это параметрический тест, который используется для получения выводов после сравнения средних для разных групп или с определенным средним для конкретной группы. Их существует несколько.

Одновыборочный t-тест  используется для сравнения среднего значения случайной выборки из набора данных со средним значением данного набора, которое уже известно.

Двухвыборочный t-тест используется для определения того, как две отдельные группы отличаются друг от друга путем сравнения их средних. Его мы и будем использовать. Данные берем за последний 2016 год.

### Средние пользовательские рейтинги платформ XOne и PC одинаковые

Нулевая гипотеза будет звучать так - средние пользовательские рейтинги по платформам XOne и PC равны.

Соответственно, альтернативная гипотеза предполагает, что они будут не равны.

In [35]:
alpha = 0.05 # критический ур.стат.значимости

user_score_XOne = last_year_data[last_year_data['platform'] == "XOne"]['user_score']
user_score_PC = last_year_data[last_year_data['platform'] == "PC"]['user_score']

results = st.ttest_ind(user_score_XOne, user_score_PC, equal_var=False, nan_policy='omit')

print('р-значение:', results.pvalue)

if results.pvalue < alpha:
    print('Отвергаем нулевую теорию')
else:
    print('Не получилось отвергнуть нулевую теорию')

р-значение: 0.35162730788932417
Не получилось отвергнуть нулевую теорию


Не получилось отвергнуть нулевую теорию. Это означает, что средние пользовательские рейтинги по платформам XOne и PC равны, либо же находятся рядом друг с другом. Это можно проверить выведя их среднии на экран.

In [36]:
print('Среднее значение XOne:', user_score_XOne.dropna().mean())
print('Среднее значение PC:', user_score_PC.dropna().mean())

Среднее значение XOne: 6.59848484848485
Среднее значение PC: 6.29777777777778


### Cредние пользовательские рейтинги жанров Action и Sports разные

Нулевая гипотеза - средние пользовательские рейтинги жанров Action и Sports равны.

Альтернативная гипотеза предполагает, что они будут не равны

In [37]:
alpha = 0.05 # критический ур.стат.значимости

user_score_action = last_year_data[last_year_data['genre'] == 'Action']['user_score']
user_score_sports = last_year_data[last_year_data['genre'] == 'Sports']['user_score']

results = st.ttest_ind(user_score_action, user_score_sports, equal_var=False, nan_policy='omit')

print('р-значение:', results.pvalue)

if results.pvalue < alpha:
    print('Отвергаем нулевую теорию')
else:
    print('Не получилось отвергнуть нулевую теорию')

р-значение: 0.0020250230280633302
Отвергаем нулевую теорию


Отвергаем нулевую теорию. Значит среднии достаточно отличаются друг от друга. Проверим.

In [38]:
print('Среднее значение жанра Action:', user_score_action.dropna().mean())
print('Среднее значение жанра Sports:', user_score_sports.dropna().mean())

Среднее значение жанра Action: 6.638823529411767
Среднее значение жанра Sports: 5.432432432432433


## Вывод

Мы проверили две гипотезы о равенстве средних между пользовательскими рейтингами платформ XOne и PC, и между пользовательскими рейтингами жанров Action и Sports. В первом случае мы не отвергли нулевую гипотезу, в отличии от второго. Соответственно, в первом случае вероятность случайно получить такое или большее различие равно около 35%. Это слишком большая вероятность, чтобы делать вывод о значимом различии между их средними.